pkg/pillar/devicenetwork/dnc.go

```diff
@@ -714,6 +717,32 @@ func (ctx *DeviceNetworkContext) doUpdatePortConfigListAndPublish(
 	return true
 }
 
+func checkAndUpdateWireless(ctx *DeviceNetworkContext, oCfg *types.DevicePortConfig, portCfg *types.DevicePortConfig) {
+	log.Infof("checkAndUpdateWireless: oCfg nil %v, portCfg Ports %v\n", oCfg == nil, portCfg.Ports)
+	for _, pCfg := range portCfg.Ports {
+		var oldPortCfg *types.NetworkPortConfig
+		if oCfg != nil {
+			for _, old := range oCfg.Ports {
+				if old.IfName == pCfg.IfName {
+					oldPortCfg = &old
+					break
+				}
+			}
+		}
+		if oldPortCfg == nil || !reflect.DeepEqual(oldPortCfg.WirelessCfg, pCfg.WirelessCfg) {
+			if pCfg.WirelessCfg.WType == types.WirelessTypeCellular ||
+				oldPortCfg != nil && oldPortCfg.WirelessCfg.WType == types.WirelessTypeCellular {
+				devPortInstallAPname(pCfg.IfName, pCfg.WirelessCfg)
+			} else if pCfg.WirelessCfg.WType == types.WirelessTypeWifi ||
+				oldPortCfg != nil && oldPortCfg.WirelessCfg.WType == types.WirelessTypeWifi {
+				if devPortInstallWifiConfig(pCfg.IfName, pCfg.WirelessCfg) {
+					ctx.WPARestartTimer = time.NewTimer(10 * time.Second)
```

Comment:
Are you sure wpa_supplicant doesn't react to a change to the file?

If it doesn't, can you make the script in the wlan container react to the change to wpa_supplicant.conf, or have it react to some other file in /run? If so you could write/touch that second file here instead of needing the timer.

Reply:
In my test, the process does not react to the change in this file.
I'll investigate the mechanism as you suggested.